src/workarounds/gpu_arrays.jl

```diff
@@ -4,11 +4,10 @@ using GPUArraysCore
 # https://github.com/JuliaGPU/CUDA.jl/issues/1565
 LinearAlgebra.dot(x::AbstractGPUArray, D::Diagonal, y::AbstractGPUArray) = x' * (D * y)
 
-function lowpass_for_symmetry!(ρ::AT, basis;
-    symmetries=basis.symmetries) where {AT <: AbstractGPUArray}
+function lowpass_for_symmetry!(ρ::AbstractGPUArray, basis; symmetries=basis.symmetries)
     all(isone, symmetries) && return ρ
     # lowpass_for_symmetry! currently uses scalar indexing, so we have to do this very ugly
     # thing for cases where ρ sits on a device (e.g. GPU)
     ρ_CPU = lowpass_for_symmetry!(to_cpu(ρ), basis; symmetries)
-    to_device(basis.architecture, ρ_CPU)
+    ρ .= to_device(basis.architecture, ρ_CPU)
```

Comment:
I was going to say this is additional work in the case of CPU, but the compiler is smart enough to recognize that `x .= x` is a no-op (actually, it looks like it calls memmove directly, and memmove is smart enough to recognize the noop)

Reply:
... and additionally this is never called for CPU arrays by the way.